crates/payload/basic/src/lib.rs

```diff
@@ -88,6 +90,8 @@ pub struct BasicPayloadJobGenerator<Client, Pool, Tasks, Builder = EthereumPaylo
     ///
     /// See [PayloadBuilder]
     builder: Builder,
+    /// Stored cached_reads for new payload jobs
+    cached_reads: Arc<RwLock<CachedReads>>,
```

Comment:
we need a new helper struct for this that:
* tracks the block hash of the tip this was computed for
* includes the CachedReads

then we need to check new_payload_job if the parent hash of the attributes matches, and only then can we use the CachedReads